Assets/Scripts/Core/Tools/UIHealthCounter.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using Player;
+using TMPro;
+using UnityEngine;
+
+namespace Core.Tools
+{
+    public class UIHealthCounter : MonoBehaviour
+    {
+        [SerializeField] private PlayerEntity _playerEntity;
+        private TextMeshProUGUI _textMeshPro;
+
+        private void Start()
+        {
+            _textMeshPro = GetComponent<TextMeshProUGUI>();
+        }
+
+        private void Update()
+        {
+            _textMeshPro.SetText(_playerEntity.Health.ToString());
```

Comment:
Updating UI is an expensive operation. This should happen on event and not on update